main.py

```diff
@@ -3,41 +3,67 @@
 import requests
 import json
 
-def create_flow(service_spec, pod_spec, flow_uuid, NEON_PROJECT_ID, NEON_FORK_FROM_BRANCH_ID, NEON_API_KEY):
-    modified_pod_spec = copy.deepcopy(pod_spec)
 
-    container = modified_pod_spec['containers'][0]
+def create_flow(service_specs: list, pod_specs: list, flow_uuid, NEON_PROJECT_ID, NEON_FORK_FROM_BRANCH_ID, NEON_API_KEY):
 
-    postgres_url = ""
-    for env in container['env']:
-        if env.get("name") == "POSTGRES":
-            postgres_url = env.get("value")
+    # main vars
+    modified_pod_specs = []
+    dev_branch_id = ""
 
-    dev_branch_hostname, dev_branch_id, error = create_neon_branch(NEON_API_KEY, NEON_PROJECT_ID, NEON_FORK_FROM_BRANCH_ID)
-    if error:
-        print(f"Error: {error}")
+    is_there_one_container_at_least: bool = False
+    for pod_spec in pod_specs:
+        modified_pod_spec = copy.deepcopy(pod_spec)
+        containers = modified_pod_spec.get('containers', [])
 
-    new_postgres_url = update_postgres_url(postgres_url, dev_branch_hostname)
-    container['env'] = [
-        {'name': 'POSTGRES', 'value': new_postgres_url},
-    ]
+        # Check if at least one container has been received otherwise the resource creation shouldn't be requested
```

Comment:
Is there every a situation where this plugin could be called without the pod spec having a container?

Reply:
@tedim52  It's just for caution: avoid creating an external resource when the manifest file contains an error.